vortex-array/src/stats/precision.rs

```diff
@@ -13,12 +13,24 @@ use crate::stats::precision::Precision::{Exact, Inexact};
 /// This is statistic specific, for max this will be an upper bound. Meaning that the actual max
```

Comment:
I made a bunch of changes in this module that I think are somewhat more idiomatic, nothing groundbreaking and I think @joseph-isaacs agreed on most.